flink-ml-lib/src/main/java/org/apache/flink/ml/common/gbt/defs/Split.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.common.gbt.defs;
+
+import org.apache.flink.api.common.typeinfo.TypeInfo;
+import org.apache.flink.ml.common.gbt.typeinfo.SplitTypeInfoFactory;
+
+import org.eclipse.collections.impl.map.mutable.primitive.IntDoubleHashMap;
+
+import java.util.BitSet;
+
+/** Stores a split on a feature. */
```

Comment:
Does this class represents the same logical concept as `org.apache.spark.mllib.tree.model.Split`? If yes, would it be more intuitive to move it to a package similar to Spark's package path instead of putting it under `gbt` package?

Reply:
If following SparkML's practice, we can add a package named `org.apache.flink.ml.tree`, and put all such classes and parameters mentioned above into this package. 

Are you OK with this?

Reply:
Sorry that I need to correct myself: the `Split` here is not a general definition: 
 - The `BinnedInstance` in `Split#shouldGoLeft(BinnedInstance binnedInstance)` is actually only applicable in GBT algorithms, as binning continuous features is a practice which normal decision trees algorithm don't do.
 - Same as `categoriesGoLeft` in `CategoricalSplit`.

I think it is better to improve these parts after more tree-based algorithms implemented.